### DISCUSSION TITLE
Nightscout - 502 Bad Gateway message

### DISCUSSION THREAD

**Discussion by kresimirjedud** [NightscoutFoundation/xDrip#2665] [General]
Hi,
I maintain 3 GC instances with FreeDNS domains. All 3 use the same domain: ftp.sh. Two are not functioning, one works just fine (therefore I conclude it is not a domain issue). All 3 worked just fine from November/22 untill a few days ago when I had a domain issue when the domain soon.it stopped functioning, so I changed to ftp.sh domain. Everything worked just fine untill this morning when two domains/subdomains stopped functioning, and one still works normal. All 3 use the latest version of Nightscout. 

I tried reinstalling the certificate, as well as installation phase 2, but no help.

I can ping and tracert all 3 addresses succesfully and I can approach one, all from the same computer. Therefore it is not a router and/or a network issue. I even tried approaching via mobile internet just to exclude the network issue. I am posting GC status and logs. I believe this is a Nightscout issue, since both the GC instance and the domain function well. I don't have any problem in reconfiguring everything, but if the community has an issue I am willing to help to find it out. One more thing - I am not under pressure to get it functioning. 

![3](https://user-images.githubusercontent.com/71585534/217620471-b6bcc2a3-8007-4296-a5d3-a63f44586b81.PNG)
![2](https://user-images.githubusercontent.com/71585534/217620475-5a801f1f-a51b-446c-8412-6bf47a61b345.PNG)
![1](https://user-images.githubusercontent.com/71585534/217620477-cb13fbba-d82b-42a5-80b7-7ccd42ef7232.PNG)

**Comment by Navid200** [comment]
The following is what Tzachi wanted to post on facebook.
  
---  
   
Here are two things that you can try:
1) run curl 127.0.0.1:1337/api/v1/entries.json
what does this return?
2) attach to the screen session:
sudo screen -r nightscout
This should show you the nightscout logs.
can you see if something is changing there?
Can you post a screen shot?
when done, you can press "Ctrl a d" to detach from the session (or you can close the ssh as well).

**Reply by kresimirjedud** [reply]
1. curl: (7) Failed to connect to 127.0.0.1 port 1337: Connection refused
2. 
![ns](https://user-images.githubusercontent.com/71585534/217624422-2fa5c9f3-8c7e-4856-955b-d3bab193a97d.PNG)

**Reply by tzachi-dar** [reply]
@kresimirjedud
Did you make any update to your system before this has happened?

Please run:
cd /srv/nightscout-vps
sudo  git log --name-status HEAD^..HEAD
sudo git status

**Reply by kresimirjedud** [reply]
@tzachi-dar I did an update a few days ago, but it all functioned after the update for a few days until this morning. I run the command ( attaching the photo) and rebooted the server butNS is still not functionin
![4](https://user-images.githubusercontent.com/71585534/217628595-2a8cea26-2f4d-4545-bdf1-5be0b7541a76.PNG)

**Reply by tzachi-dar** [reply]
Can you please run:
npm list --depth=0
and see if you lodash there?

I'm quite sure that running the commands from below should solve your issue:
```
cd /srv/nightscout-vps
npm install lodash --save
```

**Reply by kresimirjedud** [reply]
Hi,
the right command was sudo npm install lodash --save and yes, this did fix the problem and the NS is running. Do we have any conclusion? Why this happened and how to prevent it or what to do if this happens again? I'm thinking it would be wise to have instructions to run some commands each time there is an issue, and before we contact you guys, just so we save your time. Do you think that would be OK? I actually don't care what happened, just so I know how to fix it if it happens again ... 

PS - thank you for your help!

![5](https://user-images.githubusercontent.com/71585534/217639007-24a32563-3cd0-4c0d-8358-ea08101ba878.PNG)

**Reply by davidmtml** [reply]
Just wanted to add that installing lodash using these instruction fixed my 502 Gateway error as well.

Thank you!

**Comment by Kaivish** [comment]
Thank you! This solved the issue bybinstallin lodash.
But by the end I get a box mentioning about a major version of npm. Shoukd I update that?

**Reply by Navid200** [reply]
Please don't install lodash.

We are going through troubleshooting.  
After we figure out what the problem is, we will release a new issue.

If anyone experiences this problem, please run installation phase1 followed by installation phase 2.

**Reply by Kaivish** [reply]
I was a bit eager to get it fixed and read here that doing this fixed the issue. Already installed lodash 🤦🏾‍♂️. Would that result in any issues down the road?

**Reply by Navid200** [reply]
I don't think so.

There are bugs in any software, open source or not.  This is a bug and we need to fix it.
I prefer that we find what caused it and implement a solution and issue a new release so that everyone can just update to get the fix and not have to experience this problem.

But, we need a little bit of time to do that.

I don't think you will have any problems.  and even if you do, we will fix it for you.

**Reply by Kaivish** [reply]
Thank you again! Can't express how much it is appreciated the amount of help you provide .

**Comment by Navid200** [comment]
There is no bug.  There was a mistake in the guide.  I have corrected it.

If you ever run a bootstrap, you must follow by running installation phase 1.  The update platform instructions has been updated to reflect that:
https://navid200.github.io/xDrip/docs/Nightscout/NS_SyncExecutables

If you experience the bad gateway error, and if you ran bootstrap, please run installation phase 1.

**Reply by kresimirjedud** [reply]
There must be some micro constellation because doing the same thing on 3 instances and having one working fine suggests that outcome. Never the less - this is a good outcome. Thank you and everyone included 100x ... People paid to do their job don't do it like you guys do.

**Comment by Navid200** [comment]
Running bootstrap, clears a directory.  One of the contents of that directory is lodash.
Installing phase 1 brings lodash and everything else needed back.

If you installed lodash, there is no harm done.  Just run installation phase 1 from the Google Cloud setup submenu as well.
And restart the server.

**Comment by idanblum** [comment]
Hi @Navid200 ,
Thank you for the above. I encountered the 502 error and followed your instructions.
The status page no longer has missing modules and looks like this (not "NS proc:" is empty):
![image](https://user-images.githubusercontent.com/52409197/233068637-9bc4fde0-f78c-44e5-9970-1d6c1e9cd7e8.png)

If I reboot the server, it solves the issue for a few minutes and then the 502 error returns.
I will appreciate your assistance please.

**Reply by Navid200** [reply]
Did you see this?
https://github.com/NightscoutFoundation/xDrip/discussions/2788#discussioncomment-5659798

Please follow Tzachi's advice and post in that thread if it still is not resolved.

**Reply by idanblum** [reply]
Thank you @Navid200 !
I missed it before and after running it, problem seems to have been solved.
Thank you very much!!!
